packages/devtools_app/lib/src/screens/logging/logging_screen_v2/logging_table_v2.dart

```diff
@@ -0,0 +1,369 @@
+              double progress = cacheLoadProgress;
+              if (!_progressStopwatch.isRunning) {
+                _progressStopwatch.start();
+              }
+
+              if (_progressStopwatch.elapsedMilliseconds <
+                  _LoggingTableContents._millisecondsUntilCacheProgressShows) {
+                progress = 0.0;
+              }
+
+              return Center(
+                child: Column(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  children: [
+                    const Text(
+                      'Resizing... this will only take a moment.',
+                    ),
+                    const SizedBox(height: defaultSpacing),
+                    SizedBox(
+                      width: defaultLinearProgressIndicatorWidth,
+                      height: defaultLinearProgressIndicatorHeight,
+                      child: LinearProgressIndicator(
+                        value: progress,
+                      ),
+                    ),
+                    const SizedBox(height: defaultSpacing),
+                    if (_progressStopwatch.elapsedMilliseconds >
+                        _LoggingTableContents
+                            ._millisecondsUntilCacheProgressHelperShows)
+                      const Text(
+                        'To make this process faster, then reduce the "Log Retention" setting.',
+                      ),
+                  ],
+                ),
+              );
+            } else {
+              _progressStopwatch.stop();
+              _progressStopwatch.reset();
+              return Scrollbar(
+                thumbVisibility: true,
+                controller: widget._verticalController,
+                child: ListenableBuilder(
+                  listenable: widget.model,
+                  builder: (context, _) {
+                    return CustomScrollView(
+                      controller: widget._verticalController,
+                      slivers: <Widget>[
+                        SliverVariedExtentList.builder(
+                          itemCount: widget.model.logCount,
+                          itemBuilder: (context, index) {
+                            return LoggingTableRow(
+                              index: index,
+                              data: widget.model.getFilteredLog(index),
+                              isSelected: false,
+                            );
+                          },
+                          itemExtentBuilder: (index, _) =>
+                              widget.model.getFilteredLogHeight(index),
+                        ),
+                      ],
+                    );
+                  },
+                ),
+              );
+            }
+          },
+        );
+      },
+    );
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _progressStopwatch.stop();
+  }
+}
+
+class _LoggingTableContextMenu extends StatelessWidget {
```

Comment:
can we pull all the context menu code out into a follow up PR? This seems like it will need an equally thorough review and maybe some discussion, so better to do in a separate PR.